image_transport/src/republish.cpp

```diff
@@ -83,31 +120,34 @@ int main(int argc, char ** argv)
     rclcpp::SubscriptionOptions sub_options;
     sub_options.qos_overriding_options = qos_override_options;
 
-    auto sub = image_transport::create_subscription(
-      node.get(), in_topic, std::bind(pub_mem_fn, &pub, std::placeholders::_1),
+    this->sub = image_transport::create_subscription(
+      this, in_topic, std::bind(pub_mem_fn, &pub, std::placeholders::_1),
       in_transport, rmw_qos_profile_default, sub_options);
-    rclcpp::spin(node);
   } else {
     // Use one specific transport for output
-    std::string out_transport = vargv[2];
-
     // Load transport plugin
     typedef image_transport::PublisherPlugin Plugin;
     pluginlib::ClassLoader<Plugin> loader("image_transport", "image_transport::PublisherPlugin");
     std::string lookup_name = Plugin::getLookupName(out_transport);
 
     auto instance = loader.createUniqueInstance(lookup_name);
     std::shared_ptr<Plugin> pub = std::move(instance);
-    pub->advertise(node.get(), out_topic);
+    pub->advertise(this, out_topic);
```

Comment:
should this be this->pub? rather than declaring a new one in the line above? Might that be your crash?